Cura.proto

```diff
@@ -61,6 +61,23 @@ message Polygon {
     float line_width = 3; // The width of the line being laid down
 }
 
+message LayerOptimized {
+    int32 id = 1;
+    float height = 2; // Z position
+    float thickness = 3; // height of a single layer
+
+    repeated PathSegment path_segment = 4; // layer data
```

Comment:
But would the layer be flushed then? Wouldn't it be better to just send two
optimisedLayer s at the same height?

There is a feature which might just possibly be implemented at some time:
printing a couple of layers of one object before printing a couple of
layers of the other object, etc.

Anyway I think simplicity of the current code is more important than
flexibility for future possibly-might-be's.
Op 16 jun. 2016 00:29 schreef "Johan" <notifications@github.com>:

> In Cura.proto
> <https://github.com/Ultimaker/CuraEngine/pull/354#discussion_r67257940>:
>
> > @@ -61,6 +61,23 @@ message Polygon {
> >      float line_width = 3; // The width of the line being laid down
> >  }
> >
> > +message LayerOptimized {
> > +    int32 id = 1;
> > +    float height = 2; // Z position
> > +    float thickness = 3; // height of a single layer
> > +
> > +    repeated PathSegment path_segment = 4; // layer data
>
> A very good question. I think the answer is to be safe and flexibility for
> future changes. The PathCompiler flushes the data when the layer changes,
> if the layer were to jump back to a previous layer you would get two
> PathSegment messages for that layer. I've not seen this happen but I'm not
> certain that it can't happen or won't happen in the future.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/Ultimaker/CuraEngine/pull/354/files/c202d1b47a6b1539fe2b9b8c659e84d210ad4569#r67257940>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AIe9EQvTBaMkNzjYPvdQEGFq3TvFPYVaks5qMHzjgaJpZM4I1gwZ>
> .
>